### PR TITLE
Added README for corehq.apps.users [ci skip]

### PR DESCRIPTION
**Pull request by orangejenny**
Basic docs for users. Many more things could be added. Pinging USH and the newer devs for review.

### CODE REVIEW

File: corehq/apps/users/README.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+From a product perspective, there are two types of users:
+
+* Web users administrate projects. They primarily use CommCare HQ, typically on a desktop. Their tasks include building
+  applications, viewing reports, exporting data, etc. They manage their own accounts: they may create accounts
+  or may create an account by accepting an invitation, but in either case, they log in with their email address
+  and set their own password. Web users may use multiple domains and have a "membership" in each domain they access.
+* Mobile users collect data. They primarily use CommCare, typically on a mobile device. Their accounts
+  are managed by administrators, who create their accounts and set their passwords. On projects that use
+  Web Apps, "mobile" users don't use a mobile device. Mobile users are associated with a dingle domain. In code,
```
Comment (stephherbers) [inline corehq/apps/users/README.rst:15]:
```suggestion
  Web Apps, "mobile" users don't use a mobile device. Mobile users are associated with a single domain. In code, 
```

**Review by stephherbers — APPROVED** [review]
this is a great addition

**Review by esoergel — APPROVED** [review]
looks great!

File: corehq/apps/users/README.rst [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+Terminology
+~~~~~~~~~~~
+
+From a product perspective, there are two types of users:
+
+* Web users administrate projects. They primarily use CommCare HQ, typically on a desktop. Their tasks include building
+  applications, viewing reports, exporting data, etc. They manage their own accounts: they may create accounts
+  or may create an account by accepting an invitation, but in either case, they log in with their email address
+  and set their own password. Web users may use multiple domains and have a "membership" in each domain they access.
+* Mobile users collect data. They primarily use CommCare, typically on a mobile device. Their accounts
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/users/README.rst:13]: I meant to say something about users vs workers, will add that.

[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/users/README.rst:13]: e3e3d009565

**Review by gherceg — COMMENTED** [review]
🥇

**Comment by millerdev** [comment]
Thanks for adding these docs. This is very useful!

File: corehq/apps/users/README.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
+Users
+=====
+
+Terminology
+~~~~~~~~~~~
+
+From a product perspective, there are two types of users:
+
+* Web users administrate projects. They primarily use CommCare HQ, typically on a desktop. Their tasks include building
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/users/README.rst:9]: going to skip this (they're synonyms)

**Pull request merged**